### PR TITLE
feat(gatsby-plugin-styletron): update to v5, debug mode, useStyletron

### PR DESCRIPTION
**Pull request by tajo**
## Description

- update to the latest v5 styletron-react
- add [debug mode](https://www.styletron.org/api/#browser-debug-mode) when deving
- remove hacky `styletron.css()` export since now you can use a new native hook based API instead:

```js
imporet {css} from 'styletron-react';
const [css] = useStyletron();
```

**This is a breaking change, the plugin should be bumped to v4.*

### CODE REVIEW

File: packages/gatsby-plugin-styletron/src/gatsby-browser.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
 const React = require(`react`)
 const styletron = require(`./index`)
-const { Provider } = require(`styletron-react`)
+const { Provider, DebugEngine } = require(`styletron-react`)
+
+const debug = process.env.NODE_ENV === `production` ? void 0 : new DebugEngine()
```
Comment (DSchau) [inline packages/gatsby-plugin-styletron/src/gatsby-browser.js]: Could/should this also be configurable? e.g. with pluginOptions?

Comment (tajo) [reply packages/gatsby-plugin-styletron/src/gatsby-browser.js]: Yep! You can now disable it:

```js
module.exports = {
  plugins: [
    {
      resolve: `gatsby-plugin-styletron`,
      options: {
        debug: false
      },
    },
  ],
}
```

**Dev ENV:** By default, it's enabled. Can be disabled through `options`.
**Prod ENV:** Always disabled. It adds quite an overhead, so it should never be used in production.

**Comment by DSchau** [comment]
Whoops - one more comment @tajo--do we want to mention that this now requires styletron 5, e.g. like how we mention [gatsby-plugin-emotion requires emotion 10](https://www.gatsbyjs.org/packages/gatsby-plugin-emotion/#gatsby-plugin-emotion)?

**Comment by tajo** [comment]
> Whoops - one more comment @tajo--do we want to mention that this now requires styletron 5, e.g. like how we mention [gatsby-plugin-emotion requires emotion 10](https://www.gatsbyjs.org/packages/gatsby-plugin-emotion/#gatsby-plugin-emotion)?

Added a note. 

Styletron is a depedency of this plugin, so it will get bumped automatically anyway. Emotion is a peerDep.

**Comment by DSchau** [comment]
Published as `gatsby-plugin-styletron@4.0.0`. Thanks for all your help here!

**Pull request merged**